application/src/main/java/run/halo/app/core/user/service/impl/UserServiceImpl.java

```diff
@@ -210,6 +213,18 @@ public Mono<User> signUp(SignUpData signUpData) {
             .switchIfEmpty(Mono.error(() -> new ServerWebInputException(
                 "The registration is not allowed by the administrator."
             )))
+            .filter(setting -> {
+                String protectedUsernamesStr = setting.getProtectedUsernames();
+                if (protectedUsernamesStr == null || protectedUsernamesStr.trim().isEmpty()) {
+                    return true;
+                }
+                List<String> protectedList = Arrays.stream(protectedUsernamesStr.split(","))
+                        .map(String::trim)
+                        .filter(name -> !name.isEmpty())
```

Comment:
I suggest converting all names into lower case (or upper case) for subsequent comparisons.